make fails on a Mac

(**NOTE**: This is used to report product bugs:
  To report a security vulnerability, please visit <https://istio.io/about/security-vulnerabilities>
  To ask questions about how to use Istio, please visit <https://discuss.istio.io>)

**Bug description**

`make clean istioctl` is now failing on a Mac. It seems to happen on `master` and `release-1.8` branches and not `release-1.9`


[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[x] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade



**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

```
make istioctl  
ISTIO_OUT=/work/out/darwin_amd64 ISTIO_BIN=/gobin GOOS_LOCAL=darwin bin/retry.sh SSL_ERROR_SYSCALL bin/init.sh
Skipping envoy debug. Set DEBUG_IMAGE to download.
/work/out/linux_amd64/release /work
Downloading envoy: https://storage.googleapis.com/istio-build/proxy/envoy-alpha-7437add6800ef67a29cba2f276889e98dbb638a6.tar.gz to /work/out/linux_amd64/release/envoy-7437add6800ef67a29cba2f276889e98dbb638a6

real	0m4.752s
user	0m0.696s
sys	0m0.767s
Copying /work/out/linux_amd64/release/envoy-7437add6800ef67a29cba2f276889e98dbb638a6 to /work/out/linux_amd64/release/envoy
/work
/work/out/linux_amd64/release /work
Downloading envoy: https://storage.googleapis.com/istio-build/proxy/envoy-centos-alpha-7437add6800ef67a29cba2f276889e98dbb638a6.tar.gz to /work/out/linux_amd64/release/envoy-centos-7437add6800ef67a29cba2f276889e98dbb638a6

real	0m4.501s
user	0m0.677s
sys	0m0.676s
Copying /work/out/linux_amd64/release/envoy-centos-7437add6800ef67a29cba2f276889e98dbb638a6 to /work/out/linux_amd64/release/envoy-centos
/work
/work/out/darwin_amd64/release /work
Downloading envoy: https://github.com/istio/proxy/releases/download/1.0.2/istio-proxy-1.0.2-macos.tar.gz to /work/out/darwin_amd64/release/envoy-1.0.2

real	0m2.199s
user	0m0.235s
sys	0m0.206s
bin/init.sh: line 80: 3: unbound variable
Unexpected failure: Skipping envoy debug. Set DEBUG_IMAGE to download.
/work/out/linux_amd64/release /work
Downloading envoy: https://storage.googleapis.com/istio-build/proxy/envoy-alpha-7437add6800ef67a29cba2f276889e98dbb638a6.tar.gz to /work/out/linux_amd64/release/envoy-7437add6800ef67a29cba2f276889e98dbb638a6

real	0m4.752s
user	0m0.696s
sys	0m0.767s
Copying /work/out/linux_amd64/release/envoy-7437add6800ef67a29cba2f276889e98dbb638a6 to /work/out/linux_amd64/release/envoy
/work
/work/out/linux_amd64/release /work
Downloading envoy: https://storage.googleapis.com/istio-build/proxy/envoy-centos-alpha-7437add6800ef67a29cba2f276889e98dbb638a6.tar.gz to /work/out/linux_amd64/release/envoy-centos-7437add6800ef67a29cba2f276889e98dbb638a6

real	0m4.501s
user	0m0.677s
sys	0m0.676s
Copying /work/out/linux_amd64/release/envoy-centos-7437add6800ef67a29cba2f276889e98dbb638a6 to /work/out/linux_amd64/release/envoy-centos
/work
/work/out/darwin_amd64/release /work
Downloading envoy: https://github.com/istio/proxy/releases/download/1.0.2/istio-proxy-1.0.2-macos.tar.gz to /work/out/darwin_amd64/release/envoy-1.0.2

real	0m2.199s
user	0m0.235s
sys	0m0.206s
bin/init.sh: line 80: 3: unbound variable
Makefile.core.mk:224: recipe for target '/work/out/darwin_amd64/istio_is_init' failed
make[1]: *** [/work/out/darwin_amd64/istio_is_init] Error 1
make: *** [istioctl] Error 2
```

--- a/bin/init.sh
+++ b/bin/init.sh
@@ -128,7 +128,7 @@ download_envoy_if_necessary "${ISTIO_ENVOY_CENTOS_RELEASE_URL}" "$ISTIO_ENVOY_CE
 
 if [[ "$GOOS_LOCAL" == "darwin" ]]; then
   # Download and extract the Envoy macOS release binary
-  download_envoy_if_necessary "${ISTIO_ENVOY_MACOS_RELEASE_URL}" "$ISTIO_ENVOY_MACOS_RELEASE_PATH"
+  download_envoy_if_necessary "${ISTIO_ENVOY_MACOS_RELEASE_URL}" "$ISTIO_ENVOY_MACOS_RELEASE_PATH" "${SIDECAR}"
   ISTIO_ENVOY_NATIVE_PATH=${ISTIO_ENVOY_MACOS_RELEASE_PATH}
 else
   ISTIO_ENVOY_NATIVE_PATH=${ISTIO_ENVOY_LINUX_RELEASE_PATH}

